src/imginputfileconn.h

```diff
@@ -453,81 +455,92 @@ namespace dd
 
     void fillup_parameters(const APIData &ad)
     {
+      auto params = ad.createSharedDTO<dd::DTO::ImgInputConnectorParameters>();
+
       // optional parameters.
-      if (ad.has("width"))
-        _width = ad.get("width").get<int>();
-      if (ad.has("height"))
-        _height = ad.get("height").get<int>();
-      if (ad.has("crop_width"))
+      if (params->width != nullptr)
```

Comment:
Maybe we could write a short templated function that does the check against `nullptr` implicitely to simplify the code ?

Reply:
you can do just:

```cpp
if (params->width) ... //<-- check for null
```